sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextBase.java

```diff
@@ -215,7 +215,9 @@ private RxDocumentServiceRequest createQueryDocumentServiceRequest(Map<String, S
         switch (this.client.getQueryCompatibilityMode()) {
         case SqlQuery:
             SqlParameterList params = querySpec.getParameters();
-            Utils.checkStateOrThrow(params != null && params.size() > 0, "query.parameters",
+            // SqlQuerySpec::getParameters is guaranteed to return non-null SqlParameterList list
+            // hence no null check for params is necessary.
+            Utils.checkStateOrThrow(params.size() > 0, "query.parameters",
```

Comment:
anyhow. even if that's the case, why removing the additional sanity check is required?

Does Spotbug enforce this?

Reply:
yes, it was imposed by spotbug. The spotbug resolved that the following method never return a null, looking at the logic it seems true.

```java
public SqlParameterList getParameters() {
    if (this.parameters == null) {
        Collection<SqlParameter> sqlParameters = super.getCollection("parameters", SqlParameter.class);
        if (sqlParameters == null) {
            sqlParameters = new ArrayList<SqlParameter>();
        }

        this.parameters = new SqlParameterList(sqlParameters);
    }

    return this.parameters;
}
```

Reply:
ok thanks.